File: ts/src/test/Exchange/test.fetchMyTrades.ts
import assert from 'assert';
import { Exchange } from "../../../ccxt";
import testSharedMethods from './base/test.sharedMethods.js';
import testTrade from './base/test.trade.js';

async function testFetchMyTrades (exchange: Exchange, skippedProperties: object, symbol: string) {
    const method = 'fetchMyTrades';
    const trades = await exchange.fetchMyTrades (symbol);
    testSharedMethods.assertNonEmtpyArray (exchange, skippedProperties, method, trades, symbol);
    const now = exchange.milliseconds ();
    for (let i = 0; i < trades.length; i++) {
        testTrade (exchange, skippedProperties, method, trades[i], symbol, now, false);
    }
    testSharedMethods.assertTimestampOrder (exchange, method, symbol, trades);
}

export default testFetchMyTrades;

File: julia/Ccxt/test/exchange/test.fetchMyTrades.jl
using Test
using Ccxt
function testFetchMyTrades(exchange, skippedProperties, symbol)

    method = "fetchMyTrades";
    trades = Base.fetch(fetchMyTrades(exchange, symbol));
    assertNonEmtpyArray(testSharedMethods, exchange, skippedProperties, method, trades, symbol);
    now = milliseconds(exchange);
    i = 0
    while functions.ccxtruthy(functions.ccxt_lt(i, length(trades)))
        testTrade(exchange, skippedProperties, method, get(trades, i + 1, nothing), symbol, now);
        i += 1
    end
    assertTimestampOrder(testSharedMethods, exchange, method, symbol, trades);
    return true
end
